lib/glific/communications/message.ex

```diff
@@ -210,13 +209,14 @@ defmodule Glific.Communications.Message do
   @spec do_receive_message(Contact.t(), map(), atom()) :: :ok | {:error, String.t()}
   defp do_receive_message(contact, %{organization_id: organization_id} = message_params, type) do
     {:ok, contact} = Contacts.set_session_status(contact, :session)
+    group_id = get_group_id(message_params)
```

Comment:
create a function `maybe_create_group` similar to `maybe_create_contact` in file `lib/glific/contacts.ex` and then use group.id

Reply:
we still use `get_group_id` since we need to handle non maytapi provider also, but inside `get_group_id` we will be using new function `maybe_create_group`

Reply:
then we dont to create a smaller function which basically does is call a some other function and use the code directly